src/darwin/Framework/CHIPTests/MTROTAProviderTests.m

```diff
@@ -888,7 +988,472 @@ - (void)test005_DoBDXTransferAllowUpdateRequest
     // to _any_ of the above expectations.
     [self waitForExpectations:@[ announceResponseExpectation ] timeout:kTimeoutInSeconds];
 }
-#endif // ENABLE_TEST_005
+
+- (void)test007_DoBDXTransferWithTwoOTARequesters
+{
+    // In this test we do the following:
+    //
+    // 1) Check if the ota image files and raw image files required for this test exist for both OTA requestors
+    // 2) Advertise ourselves to the first OTA requestor.
+    // 3) When device1 queries for an image, pass the image path for the ota file generated beforehand as a pre-requisite for first
+    // OTA requestor 4) When device1 tries to start a bdx transfer, respond with success for the first OTA requestor
+    //    Set queryHandler to return busy with a busy delay of 30 secs and advertise ourselves to the second OTA requestor
+    // 5) When second OTA requestor queries send image available with busy delay of 30 secs indicating the provider is busy.
+    // 6) Send the data as the BDX transfer proceeds for the first OTA requestor
+    // 7) When BDX transfer ends, set up the query handler to return available for second OTA requestor and populate all the other
+    // BDX handlers
+    //    to handle BDX for the second requestor.
+    // 8) When device2 queries for an image, pass the image path for the ota file generated beforehand as a pre-requisite for second
+    // OTA requestor 9) Send the data as the BDX transfer proceeds for second OTA requestor 8) Confirm the downloaded ota image
+    // matches the raw image file that was generated before the test was run as a pre-requisite for the first OTA requestor 9) When
+    // device1 invokes ApplyUpdateRequest, respond with Proceed so that the update proceeds for the first OTA requestor 10) Wait for
+    // the app to restart and wait for the NotifyUpdateApplied message to confirm the app has updated to the new version for the
+    // first OTA requestor 11) Update the handlers for ApplyUpdateRequest and NotifyUpdateApplied for the second OTA requestor 12)
```

Comment:
Item 11 seems lke a repeat of item 7....  At this point I am completely confused about what this test is supposed to be doing.  Stopping here until this is addressed, so I have some hope of being able to evaluate (1) what the test is trying to do and (2) whether what it's doing matches that.

Reply:
discussed offline. will fix the comments